### PR TITLE
SBERDOMA-914 fix translates for share page and add setting locale via query param

### PR DESCRIPTION
**Pull request by leonid-d**
… query param

### CODE REVIEW

File: apps/condo/domains/ticket/components/ShareTicketModal.tsx [hunk 1]
```diff
@@ -162,9 +162,9 @@ const Warning = (props) => {
             `} />
 
             {ShareWarningEmailMessage}
-            {length > 0 && (intl.locale === RU_LOCALE
-                ? `… и ещё сотрудников ${length}`
-                : `… and more ${length} employee${length > 1 ? 's' : ''}`)}
+            {length > 0 && (intl.locale === EN_LOCALE
+                ? `… and more ${length} employee${length > 1 ? 's' : ''}`
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/components/ShareTicketModal.tsx:166]: add translation to ru/en locales, use FormattedMessage for string template matching

**Review by Dimitreee — CHANGES_REQUESTED** [review]

File: apps/condo/domains/ticket/components/ShareTicketModal.tsx [hunk 2]
```diff
@@ -129,6 +129,12 @@ const Warning = (props) => {
         employees: `${props.children[0]} ${props.children[1] ? (`\n${props.children[1]}`) : ''}`,
     })
     const length = props.children.length - 2
+
+    const ShareWarningEmailAndMoreMessage = intl.formatMessage({ id: 'ticket.shareWarningEmailAndMore' }, {
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/ShareTicketModal.tsx:133]: it's better to use two different messages

File: packages/@core.next/intl.jsx [hunk 3]
```diff
@@ -27,8 +27,9 @@ let getMessages = async (locale) => {
 let extractReqLocale = (req) => {
     try {
         const cookieLocale = nextCookie({ req }).locale
+        const queryLocale = req.query && req.query.locale
         const headersLocale = req.headers['accept-language'] && req.headers['accept-language'].slice(0, 2)
-        return cookieLocale || headersLocale || defaultLocale
+        return cookieLocale || queryLocale || headersLocale || defaultLocale
```
Comment (pahaz) [inline packages/@core.next/intl.jsx:32]: we have overridden extractReqLocale in _app.jsx

File: packages/@core.next/intl.jsx [hunk 4]
```diff
@@ -44,7 +44,7 @@ let getLocale = () => {
 let extractReqLocale = (req) => {
     try {
         const cookieLocale = nextCookie({ req }).locale
-        const headersLocale = req.headers['accept-language'] && req.headers['accept-language'].slice(0, 2)
+        const headersLocale = (req.query && req.query.locale) || (req.headers['accept-language'] && req.headers['accept-language'].slice(0, 2))
```
[the remark this answers is not in the document]
Comment (leonid-d) [reply packages/@core.next/intl.jsx]: No. Telegram preview can't execute js. And it seems that it doesn't know about cookies. And there is no way to change the "accept-language" header in its requests to the server.

File: apps/condo/lang/en.json [hunk 5]
```diff
@@ -461,9 +461,10 @@
   "TicketPurpose": "Application assignment",
   "ticketsTable.Number": "№",
   "ticketsTable.ResidentName": "Resident Name",
-  "ticket.shareTitle": "Ticket #{number} dated {date} has been shared with you.",
```
Comment (MrFoxPro) [inline apps/condo/lang/en.json:464]: № in Russian and # IN English?

**Review by MrFoxPro — APPROVED** [review]

**Review by Dimitreee — APPROVED** [review]

**Pull request merged**